File: examples/fib.c
#include <stdio.h>
#include <time.h>

double fib(double n) {
    if (n < 2.0) return n;
    return fib(n - 1.0) + fib(n - 2.0);
}

int main() {
    struct timespec start, end;
    clock_gettime(CLOCK_MONOTONIC, &start);

    double result = fib(35.0);
    printf("%f\n", result);

    clock_gettime(CLOCK_MONOTONIC, &end);
    double elapsed = (end.tv_sec - start.tv_sec) + (end.tv_nsec - start.tv_nsec) / 1e9;
    printf("C Suresi: %.6f\n", elapsed);
    return 0;
}

File: examples/fib.rs
use std::time::Instant;

fn fib(n: f64) -> f64 {
    if n < 2.0 {
        return n;
    }
    fib(n - 1.0) + fib(n - 2.0)
}

fn main() {
    let start = Instant::now();

    let result = fib(35.0);
    println!("{}", result);

    let elapsed = start.elapsed().as_secs_f64();
    println!("Rust Suresi: {:.6}", elapsed);
}
